core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -345,68 +344,69 @@ class KRaftMetadataCache(
     Option(_currentImage.cluster.broker(brokerId)).count(_.inControlledShutdown) == 1
   }
 
-  override def getAliveBrokers(): Iterable[BrokerMetadata] = getAliveBrokers(_currentImage)
+  override def getAliveBrokers(): util.List[BrokerMetadata] = getAliveBrokers(_currentImage)
 
-  private def getAliveBrokers(image: MetadataImage): Iterable[BrokerMetadata] = {
+  private def getAliveBrokers(image: MetadataImage): util.List[BrokerMetadata] = {
     image.cluster().brokers().values().asScala.filterNot(_.fenced()).
       map(b => new BrokerMetadata(b.id, b.rack))
-  }
+  }.toList.asJava
 
-  override def getAliveBrokerNode(brokerId: Int, listenerName: ListenerName): Option[Node] = {
-    Option(_currentImage.cluster().broker(brokerId)).filterNot(_.fenced()).
-      flatMap(_.node(listenerName.value()).toScala)
+  override def getAliveBrokerNode(brokerId: Int, listenerName: ListenerName): util.Optional[Node] = {
+    util.Optional.ofNullable(_currentImage.cluster().broker(brokerId))
+      .filter(broker => !broker.fenced())
+      .flatMap(broker => broker.node(listenerName.value()))
   }
 
-  override def getAliveBrokerNodes(listenerName: ListenerName): Seq[Node] = {
+  override def getAliveBrokerNodes(listenerName: ListenerName): util.List[Node] = {
     _currentImage.cluster().brokers().values().asScala.filterNot(_.fenced()).
```

Comment:
We should remove `asScala` from here and use `stream` instead.

Reply:
Thanks for the suggestion. I will update it tomorrow.